### DISCUSSION TITLE
I don't understand how to use this

### DISCUSSION THREAD

**Discussion by etherealsc** [ME-MarvinE/XCalendar#57] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/103879113/180397784-f693b368-a52e-4e3e-8549-652db1390b9d.png)
I followed the instructions in the "Getting Started" section after updating to the newest version, and this is what comes out. I would try and look through the sample project but it's so dense I can't process it. I just wanna be able to see the current month, use the arrows to switch months, to tap on a day to PushAsync() a form for an appointment tracker, and have days show up in a different colour if there's an appointment saved for that day. Can someone please show me how to make this work, or at least say "[this exact file] in the documentation covers the stuff you want to do"?

**Comment by ME-MarvinE** [comment, marked as answer]
Yes, the sample project's playground page is complicated. I would suggest looking at the examples as they are much easier to understand; most of them only showcase one function of the [Calendar](https://github.com/ME-MarvinE/XCalendar/wiki/Calendar-Properties) at a time.

* To switch months, you can call Calendar's 'NavigateCalendar' method. To set the command for the arrows, you can change the `ForwardsArrowCommand`, `ForwardsArrowCommandParameter`,`BackwardsArrowCommand` and `BackwardsArrowCommandParameter` properties of the `CalendarView`. 

* You can change the appearance and behaviour of a day by setting the `DayTemplate` property of the `CalendarView`. (Look at the 'Using [CalendarDayView](https://github.com/ME-MarvinE/XCalendar/wiki/CalendarDayView)' example in the sample app).

* For your use case I think the 'Event Calendar' example would provide the most help.

In terms of your problem with the dates not showing, can I see the code for your ViewModel please?

**Reply by ME-MarvinE** [reply]
The BindingContext of the page was not set to the ViewModel.

More info in [this comment](https://github.com/ME-MarvinE/XCalendar/discussions/57#discussioncomment-3206764).

**Comment by etherealsc** [comment]
```
using System;
using System.Windows.Input;
using Xamarin.Essentials;
using Xamarin.Forms;
using XCalendar.Core.Models;
namespace...
{
   public class AboutViewModel : BaseViewModel
  {
        public AboutViewModel()
        {
            Title = "About";
            OpenWebCommand = new Command(async () => await Browser.OpenAsync("https://aka.ms/xamarin-quickstart"));
        }
        public Calendar<CalendarDay> Calendar { get; set; } = new Calendar<CalendarDay>();

        public ICommand OpenWebCommand { get; }
  }
}
```

**Reply by ME-MarvinE** [reply]
Did you make sure to set the page's BindingContext to the ViewModel?

**Reply by etherealsc** [reply]
Ah. No I didn't. I'm still very new to Xamarin. 
![image](https://user-images.githubusercontent.com/103879113/180596545-8de65a87-6025-41aa-9012-6cde2c127057.png)
Sweet! 
The EventCalendar examples look like exactly what I need, I'll be looking through those proper next chance I get, thank you so much for the help :D

**Reply by ME-MarvinE** [reply]
No problem :D

Something that helped me when I was learning Xamarin Forms was knowing how similar Xamarin Forms and WPF are. If you can't find a solution to your problem or a guide to something for Xamarin Forms, try searching for it for WPF. It's been around for much longer, and has been used by more people, so  you're more likely to find an obscure bug or way to do something.

Also take a look at [compiled bindings](https://devblogs.microsoft.com/xamarin/compiled-bindings-xamarin-forms/), very useful for intellisense.
